fs/vfs/fs_rename.c

```diff
@@ -244,7 +244,7 @@ static int pseudorename(FAR const char *oldpath, FAR struct inode *oldinode,
   RELEASE_SEARCH(&newdesc);
   if (subdir != NULL)
     {
-      kmm_free(subdir);
+      lib_free(subdir);
```

Comment:
does kernel code require to call lib_free? I suppose that only librares(e.g. libc, mm, graphics...) which is compiled in both kernel and userspace need  call lib_free.

Reply:
If not define `__KERNEL__`, then `vasprintf` calls `lib_malloc` == `malloc`, but ESP32 added a new memory management mode `ESP32_SPIRAM_USER_HEAP`. In this mode, `malloc` only manages PSRAM space and `kmm_malloc` only manage SRAM space.
So we should modify all these "free" function calling.

Reply:
Is that for FLAT build only? I mean when we have single libc for kernel and apps?

Reply:
`ESP32_SPIRAM_USER_HEAP` is used both in FLAT and protect mode. When use `ESP32_SPIRAM_COMMON_HEAP` mode(FLAT only), we have single libc for kernel and apps, because in this mode, malloc == kmm_malloc.

Reply:
But `kmm_free` is remapped to `free` unless `CONFIG_MM_KERNEL_HEAP` is used, so are those configs using `CONFIG_MM_KERNEL_HEAP`?

Reply:
Yes, `ESP32_SPIRAM_USER_HEAP` select `MM_KERNEL_HEAP`: https://github.com/apache/nuttx/blob/master/arch/xtensa/src/esp32/Kconfig#L779.

Reply:
@xiaoxiang781216 @pkarashchenko Just for reference purposes, this is the PR that enables the separation of Kernel and Userspace heaps under Flat mode: https://github.com/apache/nuttx/pull/8140

Reply:
Ok, forget this possiblity.